src/js/shield.js

```diff
@@ -314,6 +314,34 @@ function getRouteDef(id) {
   };
 }
 
+/**
+ * Reformats an alphanumeric ref as Roman numerals, preserving any alphabetic
+ * suffix.
+ */
+export function romanizeRef(ref) {
+  let number = parseInt(ref, 10);
+  if (isNaN(number)) {
+    return ref;
+  }
+
+  let roman =
+    "M".repeat(number / 1000) +
+    "D".repeat((number % 1000) / 500) +
+    "C".repeat((number % 500) / 100) +
+    "L".repeat((number % 100) / 50) +
+    "X".repeat((number % 50) / 10) +
+    "V".repeat((number % 10) / 5) +
+    "I".repeat(number % 5);
+  roman = roman
+    .replace("DCCCC", "CM")
+    .replace("CCCC", "CD")
+    .replace("LXXXX", "XC")
+    .replace("XXXX", "XL")
+    .replace("VIIII", "IX")
+    .replace("IIII", "IV");
```

Comment:
There are [NPM packages](https://www.npmjs.com/package/romans) for converting to Roman numerals, which we could use instead of rolling our own. Either way, I’m thinking of this as a temporary polyfill until more browsers support numbering systems in [`Intl.NumberFormat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat).

Reply:
So `ref` has an Arabic number, but the IRL sign has a roman numeral??

Reply:
That’s correct. I’m asking for confirmation in the India Telegram chat, but so far it seems like the shield is the only context in which someone would use Roman numerals.

Reply:
Somewhere we discussed countries that use Eastern Arabic Script numerals in their highway shields and I thought the expectation was that the "as shown on the sign" value is what would go in the `ref` tag.  I'm wondering if we're doing the right thing for mapper feedback by doing the roman numeral conversion.

Reply:
That discussion started in #465 and continued in https://github.com/ZeLonewolf/openstreetmap-americana/issues/467#issuecomment-1169170636. But I think the national expressways pose a different sort of challenge. If it turns out that “II” should only appear in a shield, while “2” should appear in text, then we’d be doing a disservice to non-renderers by encouraging mappers to essentially put ASCII art into `ref`.

Reply:
Going back to the subject of this thread, any opinion on pulling in a dependency versus relying on this homegrown implementation?

Reply:
Oh, I'd defer to people that know how to program on that.

Reply:
I think this implementation is okay until we can use `Intl.NumberFormat`. Fingers crossed that it won't be long.